### PR TITLE
models and relations

### PR DESCRIPTION
**Pull request by sujoydatta**

### CODE REVIEW

**Comment by mdmahmudurrahman** [comment]
Ok

**Comment by MuhammadTamzid** [comment]
OK

File: Gemfile [hunk 1]
```diff
@@ -1,47 +1,39 @@
...
+
+gem "rails",                   "4.2.4"
+gem "public_activity"
+gem "bcrypt",                  "3.1.7"
+gem "faker",                   "1.4.2"
+gem "mysql2",                  "~> 0.3.18"
+gem "carrierwave",             "0.10.0"
+gem "mini_magick",             "3.8.0"
+gem "fog",                     "1.36.0"
+gem "will_paginate",           "3.0.7"
```
Comment (dieunb) [inline Gemfile]: [Kaminari](https://github.com/amatsuda/kaminari)

File: config/database.yml~ [hunk 2]
```diff
@@ -0,0 +1,24 @@
+common: &common
```
Comment (dieunb) [inline config/database.yml~]: don't push this file to git, pls add it into .gitignore file

File: Gemfile [hunk 3]
```diff
@@ -1,47 +1,39 @@
...
+gem "rails",                   "4.2.4"
+gem "public_activity"
+gem "bcrypt",                  "3.1.7"
+gem "faker",                   "1.4.2"
+gem "mysql2",                  "~> 0.3.18"
+gem "carrierwave",             "0.10.0"
+gem "mini_magick",             "3.8.0"
+gem "fog",                     "1.36.0"
+gem "kaminari"
+gem "bootstrap-will_paginate", "0.0.10"
```
Comment (dieunb) [inline Gemfile]: will_paginate? Should carefully!

File: db/migrate/20151230104707_create_book_photos.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class CreateBookPhotos < ActiveRecord::Migration
+  def change
+    create_table :book_photos do |t|
+      t.references :book, index: true, foreign_key: true
+      t.string :picture
```
Comment (dieunb) [inline db/migrate/20151230104707_create_book_photos.rb]: this field, i think no need. Read carrierwave to know it

**Pull request merged**